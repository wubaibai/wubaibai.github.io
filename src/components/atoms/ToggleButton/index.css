.button {
	background: var(--standard);
	padding: 4px 10px;
	border-radius: 5px;
	border: none;
	color: var(--secondary);
}

.button-reverse {
	composes: button;
	background: var(--secondary);
	color: var(--standard);
}
