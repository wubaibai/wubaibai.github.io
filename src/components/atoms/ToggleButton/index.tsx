import React from 'react';

import { useBoolean } from 'util/hook';

import style from './index.css';

interface ToggleButtonProperty {
	onClose: () => void;
	onOpen: () => void;
	closeTitle?: string;
	openTitle?: string;
}

const ToggleButton: React.FC<ToggleButtonProperty> = ({
	onClose,
	onOpen,
	closeTitle = 'Close',
	openTitle = 'Open',
}) => {
	const [isOpen, { toggle }] = useBoolean({
		onTrue: onOpen,
		onFalse: onClose,
		defaultBoolean: true,
	});

	return (
		<button type="button" className={isOpen ? style.button : style.buttonReverse} onClick={toggle}>
			{isOpen ? closeTitle : openTitle}
		</button>
	);
};

export default ToggleButton;
