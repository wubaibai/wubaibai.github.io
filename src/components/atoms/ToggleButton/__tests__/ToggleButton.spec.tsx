import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import ToggleButton from 'components/atoms/ToggleButton';

test('should have default value: Close', () => {
	const { container } = render(<ToggleButton onOpen={() => {}} onClose={() => {}} />);
	expect(container.firstChild.innerHTML).toMatch('Close');
});

// Create custom snapshot testing
test('should become Open after click', () => {
	const { container } = render(<ToggleButton onOpen={() => {}} onClose={() => {}} />);

	expect(container).toMatchSnapshot();

	// manually trigger the callback
	fireEvent.click(container.firstChild);

	expect(container.firstChild.innerHTML).toMatch('Open');
	// re-rendering would become Open
	expect(container).toMatchSnapshot();

	// manually trigger the callback
	fireEvent.click(container.firstChild);

	expect(container.firstChild.innerHTML).toMatch('Close');
	// re-rendering would become Close
	expect(container).toMatchSnapshot();
});
