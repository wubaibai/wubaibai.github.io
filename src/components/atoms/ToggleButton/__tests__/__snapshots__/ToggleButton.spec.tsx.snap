// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should become Open after click 1`] = `
<div>
  <button
    class="button"
    type="button"
  >
    Close
  </button>
</div>
`;

exports[`should become Open after click 2`] = `
<div>
  <button
    class="buttonReverse"
    type="button"
  >
    Open
  </button>
</div>
`;

exports[`should become Open after click 3`] = `
<div>
  <button
    class="button"
    type="button"
  >
    Close
  </button>
</div>
`;
